### PR TITLE
Don't pass link flags when building precompiled headers.

### PR DESCRIPTION
**Pull request by sbc100**
We were generating clang warnings for each linkflag we were injects
(i.e. each of our internal -L flags).

### CODE REVIEW

**Comment by sbc100** [comment]
We were filtering out the linker flags when compiler sources but not when compiling headers.

File: emcc.py [hunk 1]
```diff
@@ -2068,12 +2068,18 @@ def check_human_readable_list(items):
         clang_compiler = CXX
       else:
         clang_compiler = CC
+
+      def is_link_flag(flag):
+        return any(flag.startswith(x) for x in ('-l', '-L', '-Wl,'))
```
Comment (kripken) [inline emcc.py:2073]: `startswith` can receive a tuple of prefixes, so no need for the `any` I think?

File: emcc.py [hunk 2]
```diff
@@ -2084,11 +2090,6 @@ def check_human_readable_list(items):
       if shared.Settings.DEBUG_LEVEL >= 3 and not shared.Settings.WASM and not shared.Settings.WASM_BACKEND:
         newargs.append('-fno-discard-value-names')
```
Comment (kripken) [inline emcc.py:2091]: this item will not be in `compile_args` now, but it was before. is that ok?

Comment (sbc100) [reply emcc.py]: Good point.  I cleaned this up a little more.

**Pull request merged**